### PR TITLE
Remove userId from URI, Delete endpoint Scheduling, Get Matches Endpoint

### LINKED ISSUE #31: Create the delete mapping endpoint for the scheduling session (scheduling)
Create the delete mapping endpoint for the scheduling session #5 according to the REST specification table.

/matches/scheduling | DELETE | <UserId> | Body |   |   | 200 OK |   | If user likes to cancel scheduled Date

### LINKED ISSUE #84: Adapt URI by removing userId (User)
#12 , #10 and #8

-Adapt URI to the current REST specification for the user
-adapt tests

Estimate: 4h

### LINKED ISSUE #93: Endpoint for getting the matched users
#11 #3

- create endpoint that returns all the matched users
- add tests
- return activity if both users have true

Estimate: 3h

### PR DESCRIPTION
**Pull request by AleksandarRistic99**
closes #84 
closes #31 
closes #93 

- added tests

Note: Please merge the other pull request first before merging this one, this one is build up on the other and therefore has the changes that the other one has

### CODE REVIEW

**Review by SamuelBruegger — APPROVED** [review]
lgtm

**Review by SamuelBruegger — APPROVED** [review]

File: src/main/java/ch/uzh/ifi/hase/soprafs21/service/ActivityService.java [hunk 1]
```diff
@@ -93,4 +96,23 @@ public void setSwipingStatus(long activityId, String token, SwipeStatus swipeSta
 
         return generatedActivities;
     }
+
+    public List<Activity> getAllActivitiesOfUser(User user){
+        return activityRepository.findByUserSwipeStatusList_User(user);
+    }
+
+    public List<Activity> getAllActivitiesWithMatchedUsers(User user){
+        ActivityService activityService = new ActivityService(activityRepository,activityPresetRepository,userService);
```
Comment (SamuelBruegger) [inline src/main/java/ch/uzh/ifi/hase/soprafs21/service/ActivityService.java]: Is it necessary to create a new activityService? It should be possible to use the function with this.getAllActivitiesOfUser(user). I just don't know if we get a problem with the transactional annotation.

File: src/main/java/ch/uzh/ifi/hase/soprafs21/service/ActivityService.java [hunk 2]
```diff
@@ -93,4 +96,23 @@ public void setSwipingStatus(long activityId, String token, SwipeStatus swipeSta
...
+        List<Activity> allActivitiesWithMatchedUsers;
+
+        List<Activity> allActivitiesOfUser = activityService.getAllActivitiesOfUser(user);
```
Comment (Taremeh) [inline src/main/java/ch/uzh/ifi/hase/soprafs21/service/ActivityService.java]: Why do you create a ActivityService inside the ActivityService class?

How about:
```suggestion
        List<Activity> allActivitiesWithMatchedUsers;
        List<Activity> allActivitiesOfUser = getAllActivitiesOfUser(user);
```

File: src/main/java/ch/uzh/ifi/hase/soprafs21/service/ActivityService.java (hunk 1 above)
Comment (AleksandarRistic99) [reply src/main/java/ch/uzh/ifi/hase/soprafs21/service/ActivityService.java]: I have changed it to this.getAllactivitiesOfUser(user). I am also not sure about the transactional annotation, but the tests still work.

**Pull request merged**